Index: src/common/abstractsettings.cpp
===================================================================
--- src/common/abstractsettings.cpp	(revision 1309)
+++ src/common/abstractsettings.cpp	(working copy)
@@ -66,10 +66,7 @@
  */
 bool AbstractSettings::isUnixLike()
 {
-  if (QDir(QCoreApplication::applicationDirPath()).dirName() == "bin")
-    return true;
-
-  return false;
+  return true;
 }
 
 
Index: src/common/abstractsettings.h
===================================================================
--- src/common/abstractsettings.h	(revision 1315)
+++ src/common/abstractsettings.h	(working copy)
@@ -20,8 +20,8 @@
 #define ABSTRACTSETTINGS_H_
 
 #define SCHAT_UNIX_CONFIG(x) QDir::cleanPath(QDir::homePath() + "/.config/schat/" + x)
-#define SCHAT_UNIX_SHARE(x)  QDir::cleanPath(QCoreApplication::applicationDirPath() + "/../share/schat/" + x)
-#define SCHAT_UNIX_DOC(x)    QDir::cleanPath(QCoreApplication::applicationDirPath() + "/../share/doc/schat/" + x)
+#define SCHAT_UNIX_SHARE(x)  QDir::cleanPath(QString("/usr/share/schat/") + x)
+#define SCHAT_UNIX_DOC(x)    QDir::cleanPath(QString("/usr/share/doc/schat/") + x)
 
 #include <QMap>
 #include <QObject>
