Index: os/win32/engine/default.nsh
===================================================================
--- os/win32/engine/default.nsh	(revision 1415)
+++ os/win32/engine/default.nsh	(working copy)
@@ -52,11 +52,11 @@
 ${def} OPT_AUTOSTART       1
 ${def} OPT_AUTODAEMONSTART 0
 
-${def} SCHAT_QTDIR         "C:\Qt\4.7.1"
-${def} SCHAT_QT_BINDIR     "C:\Qt\4.7.1\bin"
+${def} SCHAT_QTDIR         "C:\Qt\4.7.1-x64"
+${def} SCHAT_QT_BINDIR     "C:\Qt\4.7.1-x64\bin"
 ${def} SCHAT_BINDIR        "..\..\out\release"
 ${def} SCHAT_DATADIR       "..\..\data"
 ${def} VC90_REDIST_DIR     "C:\Program Files\Microsoft Visual Studio 9.0\VC\redist\x86\Microsoft.VC90.CRT"
-${def} VC100_REDIST_DIR    "C:\Windows\system32"
+${def} VC100_REDIST_DIR    "C:"
 
 !endif /* DEFAULT_NSH_ */
Index: src/schat/schat.pri
===================================================================
--- src/schat/schat.pri	(revision 1415)
+++ src/schat/schat.pri	(working copy)
@@ -157,13 +157,12 @@
 }
 
 win32 {
-    HEADERS += update/verifythread.h
-    SOURCES += update/verifythread.cpp idle/idle_win.cpp
+  SOURCES += idle/idle_win.cpp
 }
-else {
-    DEFINES += SCHAT_NO_UPDATE
-}
 
+DEFINES += SCHAT_NO_UPDATE
+
+
 unix {
 #   LIBS += -lXss
 #   DEFINES += SCHAT_X11_XSS
