Index: src/schat/schat.pro
===================================================================
--- src/schat/schat.pro	(revision 199)
+++ src/schat/schat.pro	(working copy)
@@ -10,7 +10,7 @@
 INCLUDEPATH += . \
                ../common \
                ../schatd
-QT += network
+QT += network xml
 #CONFIG += console
 #DEFINES += SCHAT_DEBUG SCHAT_CLIENT
 DEFINES += SCHAT_CLIENT
Index: src/schatd/main.cpp
===================================================================
--- src/schatd/main.cpp	(revision 187)
+++ src/schatd/main.cpp	(working copy)
@@ -17,6 +17,7 @@
  */
 
 #include <QtCore>
+#include <unistd.h>
 
 #include "server.h"
 
@@ -35,7 +36,7 @@
   QFile pidfile(QCoreApplication::instance()->applicationDirPath() + "/schatd.pid");
   if (pidfile.open(QIODevice::WriteOnly | QIODevice::Text)) {
     QTextStream out(&pidfile);
-    out << QCoreApplication::instance()->applicationPid();
+    out << getpid();
     pidfile.close();
   }
 
