Index: app/client/js/3rdparty/jquery.timeago.js
===================================================================
--- app/client/js/3rdparty/jquery.timeago.js	(revision 544)
+++ app/client/js/3rdparty/jquery.timeago.js	(working copy)
@@ -63,7 +63,7 @@
       }
     },
     inWords: function(distanceMillis) {
-      var $l = this.settings.strings;
+      var $l = tr('timeago');
       var prefix = $l.prefixAgo;
       var suffix = $l.suffixAgo;
       if (this.settings.allowFuture) {
@@ -101,17 +101,8 @@
       if ($l.wordSeparator === undefined) { separator = " "; }
       return $.trim([prefix, words, suffix].join(separator));
     },
-    parse: function(iso8601) {
-      var s = $.trim(iso8601);
-      s = s.replace(/\.\d+/,""); // remove milliseconds
-      s = s.replace(/-/,"/").replace(/-/,"/");
-      s = s.replace(/T/," ").replace(/Z/," UTC");
-      s = s.replace(/([\+\-]\d\d)\:?(\d\d)/," $1$2"); // -04:00 -> -0400
-      return new Date(s);
-    },
     datetime: function(elem) {
-      var iso8601 = $t.isTime(elem) ? $(elem).attr("datetime") : $(elem).attr("title");
-      return $t.parse(iso8601);
+      return new Date(parseInt($(elem).attr('data-date'), 10));
     },
     isTime: function(elem) {
       // jQuery's `is()` doesn't play well with HTML5 in IE
@@ -164,8 +155,8 @@
     var $s = $t.settings;
 
     if (!isNaN(data.datetime)) {
-      if ( $s.cutoff == 0 || distance(data.datetime) < $s.cutoff) {
-        $(this).text(inWords(data.datetime));
+      if ( $s.cutoff === 0 || distance(data.datetime) < $s.cutoff) {
+        $(this).html(inWords(data.datetime));
       }
     }
     return this;
@@ -192,8 +183,4 @@
   function distance(date) {
     return (new Date().getTime() - date.getTime());
   }
-
-  // fix for IE6 suckage
-  document.createElement("abbr");
-  document.createElement("time");
 }));
