Index: src/3rdparty/singleapplication/singleapplication.pro
===================================================================
--- src/3rdparty/singleapplication/singleapplication.pro	(revision 791)
+++ src/3rdparty/singleapplication/singleapplication.pro	(working copy)
@@ -1,4 +1,3 @@
 TEMPLATE	= subdirs
 CONFIG		+= ordered
-SUBDIRS		+= src \
-		examples
+SUBDIRS		+= src
Index: src/3rdparty/singleapplication/src/singleapplication.cpp
===================================================================
--- src/3rdparty/singleapplication/src/singleapplication.cpp	(revision 791)
+++ src/3rdparty/singleapplication/src/singleapplication.cpp	(working copy)
@@ -23,15 +23,14 @@
 #include <QtCore/QByteArray>
 #include <QtCore/QCoreApplication>
 #include <QtCore/QDataStream>
+#include <QtCore/QDir>
 #include <QtCore/QHash>
 #include <QtCore/QReadWriteLock>
 #include <QtCore/QRegExp>
 #include <QtCore/QSharedMemory>
 #include <QtCore/QString>
 
-#ifdef Q_OS_WIN
-#include <qt_windows.h>
-#else
+#ifndef Q_OS_WIN
 #include <sys/types.h>
 #include <pwd.h>
 #include <unistd.h>
@@ -51,18 +50,7 @@
 		login = QLatin1String("qws");
 #endif
 #ifdef Q_OS_WIN
-		QT_WA({
-			wchar_t buffer[256];
-			DWORD bufferSize = sizeof(buffer) / sizeof(wchar_t) - 1;
-			GetUserNameW(buffer, &bufferSize);
-			login = QString::fromUtf16((ushort*)buffer);
-		},
-		{
-			char buffer[256];
-			DWORD bufferSize = sizeof(buffer) / sizeof(char) - 1;
-			GetUserNameA(buffer, &bufferSize);
-			login = QString::fromLocal8Bit(buffer);
-		});
+		login = QDir::home().dirName();
 #else
 		struct passwd* pwd = getpwuid(getuid());
 		if(pwd)
